src/include/sof/wait.h

```diff
@@ -157,4 +160,17 @@ static inline void wait_delay(uint64_t number_of_clks)
 		idelay(PLATFORM_DEFAULT_DELAY);
 }
 
+static inline int wait_for_completion_udelay(completion_t *comp, int us)
+{
+	int tick = clock_us_to_ticks(CLK_CPU, us);
+
+	wait_delay(tick);
```

Comment:
Why did you remove the tries ? and while loop ? This will now block for timeout and be horribly inefficient. We need to try an return as soon as condition completes.